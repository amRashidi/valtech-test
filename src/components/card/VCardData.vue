<script setup lang="ts">
import { computed } from "@vue/runtime-core";
import VBtn from "@/components/btn/VBtn.vue"
import VLazyImg from "@/components/image/VLazyImg"

    const props = defineProps<{intro?:boolean;title:string,subtitle:string;description:string;src:string;loading?:boolean;id?:string}>();
    const title = computed(() => props.title)
const subtitle = computed(() => props.subtitle)
const description = computed(() => props.description)
const src  = computed(() => props.src)
    const isIntro = computed(() => !!props.intro);
    const titleComponent = computed(() => isIntro.value ? "h2" : "h3")
    const cardClass = computed(() => ({card:true,"card--intro":isIntro.value,"card--loading":props.loading}))
    const cardSubtitleClasess = computed(() => ({card__subtitle:true,"skeleton":props.loading}))
const cardTitleClasess = computed(() => ({card__title:true,"skeleton":props.loading}))
const cardDescriptionClasess = computed(() => ({card__description:true,"skeleton":props.loading}))
</script>
<template>
    <article :class="cardClass" :aria-labelledby="`article-title-${title}`">
        <v-lazy-img class="card__img" :src="src" :alt="title" />
        <div class="card__wrapper">
            <div :class="cardSubtitleClasess" v-text="subtitle" />
            <component :is="titleComponent" :class="cardTitleClasess" v-text="title" :id="`article-title-${title}`" />
            <p :class="cardDescriptionClasess" v-text="description" />
            <v-btn v-if="!!props.id" class="card__action" :router="!!props.id" :to="{name:'City',params:{id:props.id}}" :aria-label="title"
                dark rounded text outline>
                read more
            </v-btn>
        </div>
    </article>
</template>
<style lang="scss" scoped>
@import "@/assets/styles/card";
</style>