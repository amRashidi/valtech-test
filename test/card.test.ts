import { mount } from '@vue/test-utils';
import VCardData from "../src/components/card/VCardData.vue";

test('mount card', () => {
    expect(VCardData).toBeTruthy()
    const card = mount(VCardData)
    expect(card.classes()).toContain("card");
})
test('card classses', async () => {
    expect(VCardData).toBeTruthy()
    const card = mount(VCardData, {
    })
    expect(card.classes()).toEqual(["card"])
    await card.setProps({ intro: true, loading: true })
    expect(card.classes()).toEqual(["card", "card--intro", "card--loading"])
})

test('card props', async () => {
    expect(VCardData).toBeTruthy()
    const card = mount(VCardData, {
        props: {
            title: "title",
            subtitle: "subtitle",
            description: "description",
            src: "src",
        }
    })
    expect(card.find(".card__title").text()).toEqual("title")
    expect(card.find(".card__subtitle").text()).toEqual("subtitle")
    expect(card.find(".card__description").text()).toEqual("description")
})
test('card props', async () => {
    expect(VCardData).toBeTruthy()
    const card = mount(VCardData, {
        props: {
            intro: false
        }
    })
    expect(card.find(".card__title").element).toMatchSnapshot()
    await card.setProps({ intro: false })
    expect(card.find(".card__title").element).toMatchSnapshot()
})