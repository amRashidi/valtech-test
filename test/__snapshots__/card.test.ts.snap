// Vitest Snapshot v1

exports[`card props 1`] = `
<h3
  class="card__title"
  data-v-ea0f52c8=""
  id="article-title-undefined"
/>
`;

exports[`card props 2`] = `
<h3
  class="card__title"
  data-v-ea0f52c8=""
  id="article-title-undefined"
/>
`;
